src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double firstSide;
+    private double secondSide;
+    private double height;
+
+    public IsoscelesTrapezoid(String color, double firstSide, double secondSide, double height) {
+        super(color);
+        this.height = height;
+        this.firstSide = firstSide;
+        this.secondSide = secondSide;
+    }
+
+    public IsoscelesTrapezoid(String color, double firstSide, double secondSide,
+                              double height, double area) {
+        super(color);
+        this.height = height;
+        this.firstSide = firstSide;
+        this.secondSide = secondSide;
+    }
+
+    @Override
+    public void findArea() {
+        if (getArea() == 0.0) {
+            setArea((firstSide + secondSide) / height);
+        } else {
+            System.out.println("The Area has already been calculated");
+        }
+    }
+
+    @Override
+    public String draw() {
+        return "Figure: IsoscelesTrapezoid, "
+                + "firstSide: " + firstSide
+                + " units, secondSide: " + secondSide
+                + " units, area: " + getArea()
+                + " sq.units, " + "height: " + height
+                + " units, color: " + getColor();
+    }
```

Comment:
it is better to use `findArea` instead of `getArea` here (after fix with return type)
so we will be confident that area will be calculated for us if it isn't
(same for other classes)